src/main/java/duke/main/Parser.java

```diff
@@ -0,0 +1,81 @@
+package duke.main;
+
+import duke.command.ByeCommand;
+import duke.command.CalendarCommand;
+import duke.command.CommandType;
+import duke.command.DeadlineCommand;
+import duke.command.EventCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.TodoCommand;
+import duke.exception.DukeException;
+import duke.exception.MissingDetailsException;
+import duke.exception.UnknownCommandException;
+import duke.task.TaskList;
+
+public class Parser {
```

Comment:
I really liked how you did your Parser by using enums and splitting clearly into different cases according to the command given. I was confused on why enums would even be needed but this helped me learn something new. Thank you :)

Reply:
Awww, thank you! I'm glad this helped you with a possible implementation of enums!